lib/Sema/CSSimplify.cpp

```diff
@@ -2096,6 +2137,16 @@ bool ConstraintSystem::repairFailures(
     return false;
   }
 
+  auto hasConversionOrRestriction = [&](ConversionRestrictionKind kind) {
+    return llvm::any_of(
+        conversionsOrFixes, [](const RestrictionOrFix possibleRestriction) {
+          auto restriction = possibleRestriction.getRestriction();
+          if (!restriction)
+            return false;
+          return restriction == ConversionRestrictionKind::DeepEquality;
```

Comment:
You probably meant to do `restriction == kind`, another way to do that would be:

```c++
if (auto restriction = correction.getRestriction())
  return restriction == kind;
return false;
```

And rename `possibleRestriction` to `correction`.